File: 2021/day1SonarSweep/day1.mjs
import {readFileSync} from "node:fs";

const lines = readFileSync("day1.txt",{encoding:"utf-8"});

//console.log(lines);

let data = lines.split("\n");

//console.log(data)

let numericData = data.map(function(s) {
    return parseInt(s)
})

console.log(numericData)

let counter = 0

for (let i=0;i<numericData.length-1;i++) {
    if (numericData[i+1]>numericData[i]) {
        //console.log("Increase")

        counter++
    }

    else {
        //console.log("Decrease")
    }
}

console.log(counter);

//Part 2

console.log("Part 2 *************************************")

let newData=[]

for (let u=0; u< numericData.length-2;u++) {

    let sum=numericData[u] + numericData[u+1] + numericData[u+2];

    newData.push(sum);

}

console.log(newData)

let counter2 = 0

for (let x=0;x<newData.length-1;x++) {
    if (newData[x+1]>newData[x]) {
        console.log("Increase")

        counter2++
    }

    else {
        console.log("Decrease")
    }
}


console.log(counter2);

File: 2021/day1SonarSweep/day1.py
# --- Day 1: Sonar Sweep ---

input = "day1Input.txt"

with open(input, "r", encoding="utf-8") as file:
    report = [int(m) for m in file.read().split("\n")]

larger_measurements = 0

for i in range(1, len(report)):
    if report[i] > report[i - 1]:
        larger_measurements += 1

print(
    f"The number of measurements that are larger than the previous measurement is: {larger_measurements}"
)  # Answer Part 1

# Part 2

three_measurement_list = [
    m + report[i + 1] + report[i + 2]
    for i, m in enumerate(report)
    if i < len(report) - 2
]

larger_sum_measurements = 0

for i in range(1, len(three_measurement_list)):
    if three_measurement_list[i] > three_measurement_list[i - 1]:
        larger_sum_measurements += 1

print(
    f"The number of sum measurements that are larger than the previous sum measurement is: {larger_sum_measurements}"
)  # Answer Part 2
